src/main/java/mate/academy/bookshop/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.bookshop.repository;
+
+import java.util.List;
+import mate.academy.bookshop.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.criteria.JpaCriteriaQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book into DB : " + book, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List findAll() {
```

Comment:
Don't forget to use generics with collections, it's very important, because otherwise you can get an unexpected `ClassCastException`
```suggestion
    public List<Book> findAll() {
```